test/replication-luatest/gh_7377_bootstrap_connection_failure_test.lua

```diff
@@ -35,14 +35,14 @@ g.before_all(function(cg)
     t.assert(cg.proxy:start{force = true}, 'Proxy started successfully')
     for i = 1, 3 do
         box_cfg.instance_uuid = cg.uuids[i]
-        if i == 3 then
-            box_cfg.replication[1] = server.build_listen_uri('server_1_proxy')
-        end
         cg.servers[i] = cg.cluster:build_and_add_server({
             alias = 'server_' .. i,
             box_cfg = box_cfg,
         })
     end
+    cg.servers[2].box_cfg.replication[3] = nil
+    cg.servers[3].box_cfg.replication[1] =
+        server.build_listen_uri('server_1_proxy')
```

Comment:
Why is this change needed? Maybe better just keep it `legacy` in all the old tests?

Reply:
I wanted to make it work without the `legacy` bootstrap strategy, but ok:
```diff
diff --git a/test/replication-luatest/gh_7377_bootstrap_connection_failure_test.lua b/test/replicati>
index 14751b967..26e896ade 100644
--- a/test/replication-luatest/gh_7377_bootstrap_connection_failure_test.lua
+++ b/test/replication-luatest/gh_7377_bootstrap_connection_failure_test.lua
@@ -20,6 +20,7 @@ g.before_all(function(cg)
             server.build_listen_uri('server_3'),
         },
         replication_timeout = timeout,
+        bootstrap_strategy = 'legacy',
     }
     cg.uuids = {
         'bbbbbbbb-bbbb-bbbb-bbbb-bbbbbbbbbbbb',
@@ -35,14 +36,14 @@ g.before_all(function(cg)
     t.assert(cg.proxy:start{force = true}, 'Proxy started successfully')
     for i = 1, 3 do
         box_cfg.instance_uuid = cg.uuids[i]
+        if i == 3 then
+            box_cfg.replication[1] = server.build_listen_uri('server_1_proxy')
+        end
         cg.servers[i] = cg.cluster:build_and_add_server({
             alias = 'server_' .. i,
             box_cfg = box_cfg,
         })
     end
-    cg.servers[2].box_cfg.replication[3] = nil
-    cg.servers[3].box_cfg.replication[1] =
-        server.build_listen_uri('server_1_proxy')
 end)
 
 g.after_all(function(cg)

```